core/src/main/scala/shapeless/ops/coproduct.scala

```diff
@@ -1268,4 +1268,30 @@ object coproduct {
         def apply(): Out = wh.value :: rt()
       }
   }
+
+  sealed trait LiftAll[F[_], In <: Coproduct] {
```

Comment:
The main reason for not extending `DepFn0` was that I didn't know it existed. Probably want the sam to be treatment on `HList`